pkg/cmd/historical_logs.go

```diff
@@ -107,40 +118,317 @@ func (o *LogParameters) Execute(streams genericclioptions.IOStreams) error {
+		case 'h':
+			startTime = endTime.Add(time.Duration(-tail) * time.Hour).UTC()
+		case 'd':
+			startTime = endTime.Add(time.Duration(-tail) * time.Hour * 24)
+		default:
+			return fmt.Errorf("invalid time unit entered in \"tail\". please enter s, m, h, or d as time unit")
+		}
+
+		o.StartTime = startTime.UTC().Format(time.RFC3339Nano)
+		o.EndTime = endTime.UTC().Format(time.RFC3339Nano)
 	}
 
-	client := &http.Client{}
-	res, err := client.Do(req)
-	if err != nil {
-		return fmt.Errorf("Response Error: %v", err)
+	var podList []string
+
+	if len(o.Deployment) > 0 {
+		err = GetDeploymentPodsList(clientset, o.Deployment, o.Namespace, &podList)
+		if err != nil {
+			return err
+		}
+	}
+
+	if len(o.DaemonSet) > 0 {
+		err = GetDaemonSetPodsList(clientset, o.DaemonSet, o.Namespace, &podList)
+		if err != nil {
+			return err
+		}
 	}
 
-	responseBody, err := ioutil.ReadAll(res.Body)
+	if len(o.StatefulSet) > 0 {
+		err = GetStatefulSetPodsList(clientset, o.StatefulSet, o.Namespace, &podList)
+		if err != nil {
+			return err
+		}
+	}
+
+	var logList []string
+	if len(podList) > 0 {
+		for _, pod := range podList {
+
+			o.Podname = pod
+			response,err := o.makeHttpRequest(ApiUrl)
+
+			if err!=nil{
+				return err
+			}
+
+			responseBody, err := ioutil.ReadAll(response.Body)
+			if err != nil {
+				return fmt.Errorf("failed to read response: %v", err)
+			}
+
+			err = getLogs(responseBody, &logList)
+			if err != nil {
+				return err
+			}
+		}
+
+		err = printLogs(logList, streams, o.Limit)
+		if err != nil {
+			return err
+		}
+
+	} else {
+
+		response,err := o.makeHttpRequest(ApiUrl)
+		if err!=nil{
+			return err
+		}
+		responseBody, err := ioutil.ReadAll(response.Body)
+		if err != nil {
+			return fmt.Errorf("failed to read response: %v", err)
+		}
+
+		err = getLogs(responseBody, &logList)
+		if err != nil {
+			return err
+		}
+
+		err = printLogs(logList, streams, o.Limit)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (o *LogParameters) makeHttpRequest(baseUrl string) (*http.Response,error) {
+
+	var urlBuilder strings.Builder
+	urlBuilder.WriteString(baseUrl)
+	numParameters := 0
+	if len(o.Podname) >0{
+		urlBuilder.WriteString("?")
+		urlBuilder.WriteString("podname="+o.Podname)
+		numParameters = numParameters + 1
+	}
+	if len(o.Namespace) >0{
+		if numParameters == 0{
+			urlBuilder.WriteString("?")
+		}else{
+			urlBuilder.WriteString("&")
+		}
+		urlBuilder.WriteString("namespace="+o.Namespace)
+		numParameters = numParameters + 1
+	}
+	if len(o.Limit) >0{
+		if numParameters == 0{
+			urlBuilder.WriteString("?")
+		}else{
+			urlBuilder.WriteString("&")
+		}
+		urlBuilder.WriteString("maxlogs="+o.Limit)
+		numParameters = numParameters + 1
+	}
+	if len(o.Level) >0{
+		if numParameters == 0{
+			urlBuilder.WriteString("?")
+		}else{
+			urlBuilder.WriteString("&")
+		}
+		urlBuilder.WriteString("level="+o.Level)
+		numParameters = numParameters + 1
+	}
+	if len(o.StartTime)>0 && len(o.EndTime)>0{
```

Comment:
I feel a little formatting is required too like `if len(o.StartTime) > 0 && len(o.EndTime) > 0 {`

Reply:
Done

Reply:
Done